### ISSUE TITLE
Basic auth credentials from repo not used for dependency update command

### ISSUE THREAD

**Issue by dbluxo** [helm/helm#4445]
I added a repo (Chartmuseum) with --username and --password params.
When I run `helm repo update` and `helm fetch <repo>/<chart>` everythings works fine, but for `helm dependency update` it doesn't pick up the credentials:

```
$ helm dependency update
Hang tight while we grab the latest from your chart repositories...
...Successfully got an update from the "local" chart repository
...Successfully got an update from the "chartmuseum" chart repository
Update Complete. ⎈Happy Helming!⎈
Saving 1 charts
Downloading mongodb-replicaset from repo https://chartmuseum-customurl
Save error occurred:  could not download https://chartmuseum-customurl/charts/mongodb-replicaset-3.5.0.tgz: Failed to fetch https://chartmuseum-customurl/charts/mongodb-replicaset-3.5.0.tgz : 401 Unauthorized
Deleting newly downloaded charts, restoring pre-update state
Error: could not download https://chartmuseum-customurl/charts/mongodb-replicaset-3.5.0.tgz: Failed to fetch https://chartmuseum-customurl/charts/mongodb-replicaset-3.5.0.tgz : 401 Unauthorized
```

Output of `helm version`:
```
$ helm version
Client: &version.Version{SemVer:"v2.10.0-rc.2", GitCommit:"56154102a2f25ebf679c791907fd355bb0377f05", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.10.0-rc.2", GitCommit:"56154102a2f25ebf679c791907fd355bb0377f05", GitTreeState:"clean"}
```
(Also doesn't work with 2.9.1)

Output of `kubectl version`:
```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.5", GitCommit:"f01a2bf98249a4db383560443a59bed0c13575df", GitTreeState:"clean", BuildDate:"2018-03-19T15:59:24Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"9+", GitVersion:"v1.9.5-rancher1", GitCommit:"f11c6299ce2b927c3e34ea2afdf57cd08596802f", GitTreeState:"clean", BuildDate:"2018-03-20T16:40:55Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
```
Cloud Provider/Platform (AKS, GKE, Minikube etc.):
AWS + Rancher + Kubernetes

Opened from discussion of: https://github.com/helm/helm/issues/3858

**Comment by steven-sheehy** [comment]
Related to #4299

**Comment by deyceg** [comment]
Does the JFrog version work as expected and this is just a bad implementation of that functionality?

**Comment by dbluxo** [comment]
@bartvanhoutte @bacongobbler

My workaround is to add the credentials in the repo URL manually.
You can add them here: ~/.helm/repository/repositories.yaml
Just alter the URL field:
`url: https://username:password@chartmuseum-customurl`

**Comment by deyceg** [comment]
@dbluxo how does this work? whenever I try to run `helm dependency build/update` it complains about not finding the repo in the `repositories.yaml` which I can only assume its doing some very basic regex on the file using the `requirements.yaml` `url`

**Comment by dbluxo** [comment]
@deyceg 
My repositories.yaml looks like:
```
apiVersion: v1                                                                                        
generated: 2018-06-19T14:30:42.142837882+02:00                                                        
repositories:                                                                                                                                                                                 
- caFile: ""                                                                                          
  cache: /home/user/.helm/repository/cache/chartmuseum-index.yaml                                   
  certFile: ""                                                                                        
  keyFile: ""                                                                                         
  name: chartmuseum                                                                                   
  password: password                                                                      
  url: https://username:password@chartmuseum-customurl            
  username: username
 ```

No changes in `Chart.yaml`. But you have to add the alias in the `requirements.yaml`:
```
dependencies:                      
  - name: mongodb-replicaset       
    repository: alias:chartmuseum  
    version: '>= 3.6.0'            
```

**Comment by deyceg** [comment]
@dbluxo thanks that was the missing bit (the alias)! Let me give it a whirl :)

**Comment by deyceg** [comment]
I don't think this solution works with Artifactory :( Still get the 401 unauthorized. I'm going to need to fix the chart down loader in a custom Helm build as its definitely not getting credentials added.

**Comment by deyceg** [comment]
turns out it was just a reverse proxy configuration 👍

**Comment by bacongobbler** [comment]
closed via #4662
